src/main/app/Resources/modules/security/registration/components/main.jsx

```diff
@@ -39,16 +55,52 @@ class RegistrationMain extends Component {
         title: 'Configuration',
         component: Optional
       }
-    ], this.props.facets.map(facet => ({
-      title: facet.title,
-      component: () => {
-        const currentFacet = <Facet facet={facet}/>
+    ], this.props.facets.map((facet) => ({
+      ...facet,
+      sections:
+        facet.sections !== undefined
+          ? [
+            ...facet.sections.map((section) => ({
+              ...section,
+              fields: section.fields.filter(
```

Comment:
You should use the `displayed` prop of the field to avoid rebuilding the full list of fields at each change (which will cause react to re render more things).